### PR TITLE
* replaced Object.values & Array.prototype.find with ES5 equivalents …

### PR DESCRIPTION
**Pull request by Quentie**
…(so no polyfills are necessary for old js engines e.g.: I try to use it in duktape js running on top of native c engine)

* (Bublé only transpiles language features – it doesn't attempt to shim or polyfill the environment)

### CODE REVIEW

File: src/ecs.js [hunk 1]
```diff
@@ -26,7 +26,8 @@ const matchEntity = (entity) => {
 };
 
 const ejectEntity = (entity) => {
-  Object.values(entity.components).forEach(
```
Comment (kutuluk) [inline src/ecs.js:29]: Could it be better this way?
```javascript
for (const key in entity) {
    if (entity.hasOwnProperty(key)) {
        const component = entity[key];
        component && component.destructor && component.destructor();
    }
}
```

File: src/ecs.js [hunk 2]
```diff
@@ -119,7 +120,8 @@ class Selector {
     this.list = null;
     this.length = 0;
 
-    Object.values(entities).forEach(entity => this.match(entity));
```
Comment (kutuluk) [inline src/ecs.js:122]: And here too
```javascript
for (const key in entities) {
    if (entities.hasOwnProperty(key)) {
        this.match(entities[key]);
    }
}
```

**Review by kutuluk — CHANGES_REQUESTED** [review]
Maybe at the same time improve performance? =)

File: src/ecs.js (hunk 1 above)
Comment (Quentie) [reply src/ecs.js:29]: Looks  really good to me, but your eslint configuration does not seem to like it (I have no  clue about speed comparison between both versions I have to admit).

Actually I first tried for...of loops but also eslint did complain ;)

"  29:3   error  for..in loops iterate over the entire prototype chain, which is virtually never what you want. Use Object.{keys,values,entries}, and iterate over the resulting array  no-restricted-syntax
  30:16  error  Do not access Object.prototype method 'hasOwnProperty' from target object                                                                                              no-prototype-builtins
"

Comment (kutuluk) [reply src/ecs.js:29]: Try this
```javascript
// eslint-disable-next-line no-restricted-syntax
for (const key in entity) {
    if (Object.prototype.hasOwnProperty.call(entity, key)) {
        const component = entity[key];
        component && component.destructor && component.destructor();
    }
}
```

Comment (Quentie) [reply src/ecs.js:29]: @kutuluk Sure, I have updated the pull request.

File: src/ecs.js (hunk 2 above)
Comment (Quentie) [reply src/ecs.js:122]: done

**Review by kutuluk — APPROVED** [review]

**Comment by kutuluk** [comment]
https://jsperf.com/object-keys-vs-hasownproperty/84

Perhaps your original way was the best. I'll think about which one to prefer. Thank you!

**Pull request merged**